geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/geotools/SftArgResolver.scala

```diff
@@ -22,84 +23,101 @@ import scala.util.{Failure, Success, Try}
  * including sft strings (e.g. name:String,age:Integer,*geom:Point)
  * and typesafe config.
  */
-object SftArgResolver extends LazyLogging {
+case class SftArgs(specArg: String, featureName: String)
+object SftArgResolver extends AbstractArgResolver[SimpleFeatureType, SftArgs] with LazyLogging {
 
-  // Important to setAllowMissing to false bc else you'll get a config but it will be empty
-  val parseOpts =
-    ConfigParseOptions.defaults()
-      .setAllowMissing(false)
-      .setClassLoader(null)
-      .setIncluder(null)
-      .setOriginDescription(null)
-      .setSyntax(null)
+  import ArgTypes._
 
-  /**
-   * @return the SFT parsed from the Args
-   */
-  def getSft(specArg: String, featureName: String = null): Option[SimpleFeatureType] =
-    getLoadedSft(specArg, featureName)
-      .orElse(parseSpecString(specArg, featureName))
-      .orElse(parseConfStr(specArg, featureName))
-      .orElse(parseSpecStringFile(specArg, featureName))
-      .orElse(parseConfFile(specArg, featureName))
+  override def argType(args: SftArgs) = {
+    /**
+     * Here we use rudimentary checking to guess as what kind of specArg was passed in.
+     * We use this to decide which error message to display to the user, since the
+     * parsers fail frequently. The rest of the errors are logged.
+     */
+    val fileNameReg = """([^.]*)\.([^.]*)""" // e.g. "foo.bar"
+    val specStrReg = """^[a-zA-Z0-9]+[:][String|Integer|Double|Point|Date|Map|List].*""" // e.g. "foo:String..."
+    val specStrRegWError = """^[a-zA-Z0-9]+[:][a-zA-Z0-9]+.*""" // e.g. "foo:Sbartring..."
+    args.specArg match {
+      // Order is important here
+      case s if s.contains("geomesa{")
```

Comment:
wondering if this should be something along the lines of having some trait with classes for each that can have some methods like canProces(arg: String), process(arg:String): Try